Add Spark connector reader support. (#11823)
apache/druid: docs/operations/spark.md

|Key|Description|Required|Default|
|---|-----------|--------|-------|
|`broker.host`|The hostname of a broker in the Druid cluster to read from|No|`localhost`|
|`broker.port`|The port of the broker in the Druid cluster to read from|No|8082|
|`broker.numRetries`|The number of times to retry a timed-out segment metadata request|No|5|
|`broker.retryWaitSeconds`|How long (in seconds) to wait before retrying a timed-out segment metadata request|No|5|
|`broker.timeoutMilliseconds`|How long (in milliseconds) to wait before timing out a segment metadata request|No|300000|

### Reader Configs
The properties used to configure the DataSourceReader when reading data from Druid in Spark.

|Key|Description|Required|Default|
|---|-----------|--------|-------|
|`table`|The Druid data source to read from|Yes||
|`reader.deepStorageType`|The type of deep storage used to back the target Druid cluster|No|`local`|
|`reader.segments`|A hard-coded list of Druid segments to read. If set, the table and druid client configurations are ignored and the specified segments are read directly. Must be deserializable into Druid DataSegment instances|No|

> |`reader.useCompactSketches`|Controls whether or not compact representations of complex metrics are used (only for metrics that support compact forms)|No|False|

Reviewer: Then, perhaps better rephrasing this to something like: 

```suggestion
|`reader.useCompactSketches`|Controls whether or not compact representations of sketch metrics are used|No|False|
```
